staging/src/k8s.io/mount-utils/mount_helper_common.go

```diff
@@ -53,7 +53,7 @@ func CleanupMountWithForce(mountPath string, mounter MounterForceUnmounter, exte
 	}
 	var notMnt bool
 	var err error
-	if !corruptedMnt {
+	if !mounter.canSafelySkipMountPointCheck() && !corruptedMnt {
```

Comment:
to make it easy to reason the code, I kind of feel we could reduce the number of times to check canSafelySkipMountPointCheck, 

if mounter.canSafelySkipMountPointCheck() {
    // UnmountwithForce()
    // removePath if no error
} else {
   // the old logic
}